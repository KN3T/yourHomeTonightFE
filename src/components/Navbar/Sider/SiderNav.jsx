import { BarsOutlined, HomeOutlined, UserOutlined } from '@ant-design/icons';
import { Divider, Space } from 'antd';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { NavLink } from 'react-router-dom';

import './index.scss';

const SiderNav = () => {
  const userData = window.localStorage.getItem('userData')
    ? JSON.parse(window.localStorage.getItem('userData'))
    : '';

  window.addEventListener('scroll', () => {
    if (window.pageYOffset > 300) {
      const navbar = document.querySelector('.navbar__logo__wrapper');
      navbar.classList.add('navbar__logo__wrapper__with_shadow');
    } else {
      const navbar = document.querySelector('.navbar__logo__wrapper');
      navbar.classList.remove('navbar__logo__wrapper__with_shadow');
    }
  });

  const { t } = useTranslation();

  return (
    <>
      <div className="navbar__sider__wrapper">
        <Space className="space">
          <a className="link">
            <BarsOutlined className="icon" />
          </a>
        </Space>
        <Divider
          className="navbar__sider__divider"
          style={{ margin: '14px 0' }}
        />
        {userData.token ? (
          <Space className="space">
            <NavLink to="/userprofile" className="link">
              <UserOutlined className="icon" />
              <span className="text__icon">{t('navbar.profile')}</span>
            </NavLink>
          </Space>
        ) : (
          <Space className="space">
            <NavLink className="link" to="/login">
              <UserOutlined className="icon" />
              <span className="text__icon">{t('navbar.login')}</span>
            </NavLink>
          </Space>
        )}

        <Space className="space">
          <NavLink className="link" to="/">
            <HomeOutlined className="icon" />
            <span className="text__icon">{t('navbar.home')}</span>
          </NavLink>
        </Space>
      </div>
      <div className="navbar__logo__wrapper">
        <NavLink to="/">
          <h1>
            yourhome<span>29.</span>{' '}
          </h1>
        </NavLink>
      </div>
    </>
  );
};

export default SiderNav;
