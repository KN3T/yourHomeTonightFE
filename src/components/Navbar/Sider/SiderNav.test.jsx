import { render } from '@testing-library/react';
import { HashRouter } from 'react-router-dom';

import SiderNav from './SiderNav';

describe('Sider navigation test', () => {
  it('should render correctly', () => {
    const SiderNavEl = render(
      <HashRouter>
        <SiderNav />
      </HashRouter>
    );
    expect(SiderNavEl).toMatchSnapshot();
  });
});
