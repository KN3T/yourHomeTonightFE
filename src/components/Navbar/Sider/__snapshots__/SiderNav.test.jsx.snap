// Vitest Snapshot v1

exports[`Sider navigation test > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navbar__sider__wrapper"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center space"
        >
          <div
            class="ant-space-item"
          >
            <a
              class="link"
              href="#/login"
            >
              <span
                aria-label="user"
                class="anticon anticon-user icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="user"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                  />
                </svg>
              </span>
              <span
                class="text__icon"
              >
                navbar.login
              </span>
            </a>
          </div>
        </div>
        <div
          class="ant-divider ant-divider-horizontal navbar__sider__divider"
          role="separator"
          style="margin: 14px 0px;"
        />
        <div
          class="ant-space ant-space-horizontal ant-space-align-center space"
        >
          <div
            class="ant-space-item"
          >
            <a
              aria-current="page"
              class="link active"
              href="#/"
            >
              <span
                aria-label="home"
                class="anticon anticon-home icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="home"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                  />
                </svg>
              </span>
              <span
                class="text__icon"
              >
                navbar.home
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        class="navbar__logo__wrapper"
      >
        <a
          aria-current="page"
          class="active"
          href="#/"
        >
          <h1>
            yourhome
            <span>
              29.
            </span>
             
          </h1>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="navbar__sider__wrapper"
    >
      <div
        class="ant-space ant-space-horizontal ant-space-align-center space"
      >
        <div
          class="ant-space-item"
        >
          <a
            class="link"
            href="#/login"
          >
            <span
              aria-label="user"
              class="anticon anticon-user icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="user"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                />
              </svg>
            </span>
            <span
              class="text__icon"
            >
              navbar.login
            </span>
          </a>
        </div>
      </div>
      <div
        class="ant-divider ant-divider-horizontal navbar__sider__divider"
        role="separator"
        style="margin: 14px 0px;"
      />
      <div
        class="ant-space ant-space-horizontal ant-space-align-center space"
      >
        <div
          class="ant-space-item"
        >
          <a
            aria-current="page"
            class="link active"
            href="#/"
          >
            <span
              aria-label="home"
              class="anticon anticon-home icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="home"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                />
              </svg>
            </span>
            <span
              class="text__icon"
            >
              navbar.home
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      class="navbar__logo__wrapper"
    >
      <a
        aria-current="page"
        class="active"
        href="#/"
      >
        <h1>
          yourhome
          <span>
            29.
          </span>
           
        </h1>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
